### ISSUE TITLE
move navcons into unvanquished.git repo

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#2730]
Currently, we have 11 maps. Each time a new feature or improvement related to navcon code is done, one would have to create 11 PRs to update all official maps, which is quite the work for not much.

Navcon files are directly related to 4 things:

* navmesh, which is generated dynamically by the C++ code
* navcon parser, implemented by the C++ code
* agent's sizes, which are defined in res-players (sadly, imo those should be moved in the unvanquished_src.dpkdir config files, would make things easier)
* agent's abilities, which are defined int unvanquished_src.dpkdir config files

Also, at some point I hope to be able to generate at least some of them automatically.

Considering those elements, I think it would make sense and reduce the maintenance burden to move them directly in this repo instead of having them scattered around many places.

What do you think about this idea, @sweet235, @illwieckz ?

**Comment by illwieckz** [comment]
I believe it's a bad idea. The navcons files are map specific, we can see them as part of the bsp, so they mus be as close as the bsp. When a map is significantly updated (like the Parpax revamp we seen), having the navcons in another dpk is the best way to break things.

**Comment by bmorel** [comment]
They are not reliated to bsp, but to navmeshes, bboxes and class abilities such as wallwalking or being able to use ladders.

**Comment by bmorel** [comment]
Is not there a DELETED or alike feature in the .dpk DEPS file that could be used to prevent that problem?

**Comment by slipher** [comment]
No. You can think of the VFS as a map of `string (filename) -> optional<string> (file contents)`. The first entry for a given key is kept and subsequent ones are ignored. Having a file in the dpk populates the entry with the string contents and having it in DELETED populates it with an empty optional. So DELETED does not allow you to go against the loading order of paks.

**Comment by bmorel** [comment]
I see. Well, if this is not technically feasible anyway, let's close.
